datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/validators/validator.py

```diff
@@ -0,0 +1,198 @@
+    """
+    def __init__(self):
+        self.result = ValidationResult()
+
+    def validate(self, profile, directory, path):
+        # type: (str, str, list(str)) -> None
+        raise NotImplementedError
+
+    def fail(self, error_message):
+        self.result.failed = True
+        self.result.messages['failure'].append(error_message)
+
+    def fix(self, problem, solution):
+        self.result.warning_msg = problem
+        self.result.success_msg = solution
+        self.result.fixed = True
+        self.result.failed = False
+
+    def success(self, success_message):
+        self.result.messages['success'].append(success_message)
+    
+    def info(self, info_message):
+        self.result.messages['info'].append(info_message)
+    
+    def warning(self, warning_message):
+        self.result.messages['warning'].append(warning_message)
+
+    def __repr__(self):
+        return str(self.result)
+
+class SchemaValidator(ProfileValidator):
+    """"
+    Validator responsible to check if the profile matches with the schemas.
+    """
+    def __init__(self):
+        super(SchemaValidator, self).__init__()
+        self.errors = []
+        self.contents = None
+
+    def __repr__(self):
+        return self.file_path
+
+    def load_from_file(self, file_path,path):
+        self.contents = find_profile_in_path(file_path,path, line = False)
+        if not self.contents:
+            self.fail("File contents returned None: " + file_path)
+            
+
+    def validate(self, profile: str, directory: str, path: list):
+        schema_file = join(
+            get_root(),
+            "datadog_checks_dev",
+            "datadog_checks",
+            "dev",
+            "tooling",
+            "commands",
+            "meta",
+            "snmp",
+            "validators",
+            "profile_schema.json",
+        )
+        self.load_from_file(profile,path)
+
+        with open(schema_file, "r") as f:
+            contents = f.read()
+            schema = json.loads(contents)
+        validator = jsonschema.Draft7Validator(schema)
+
+        errors = validator.iter_errors(self.contents)
+        for error in errors:
+            self.errors.append(error)
+            self.fail(error.message)
+        
+        if len(self.errors) == 0:
+            self.success("Schema successfully validated")
+
+class DuplicateMetricsValidator(ProfileValidator):
+    """"
+    Validator responsible to check if there are no duplicated metrics in the profile.
+    It checks all the profiles extended by the profile passed.
+    """
+    def validate(self, profile, directory, path):
+        #type: (ProfileValidator,str, str, str) -> None
+        """
+        Calls the recursive function(verify_duplicate_metrics_profile_recursive) to check if there are any duplicated metric.
+        It also logs the duplicated OID and reports all the files:lines where it is.
+        """
+        used_metrics = {}
+        duplicated = {}
```

Comment:
Any reason not having `used_metrics` and `duplicated` as instance variable (`self.used_metrics` and  `self. duplicated`) instead of passing them down multiple times ?

Reply:
Not really. It is better that I use instance variables.
Changed